src/training/ModelFiller.cpp

```diff
@@ -1,5 +1,12 @@
 #include <ecto/ecto.hpp>
-#include <opencv2/core/core.hpp>
+
+
+#if CV_VERSION_MAJOR < 3
```

Comment:
the proper way is to look at the existence of CV_VERSION_EPOCH or not (it's not in OpenCV3) and yes, that needs to be changed in all OpenCV ROS code (for now, I create a pre-processor variable OPENCV3)